addons/mail/models/ir_http.py

```diff
@@ -13,7 +13,14 @@ def session_info(self):
         """Override to add the current user data (partner or guest) if applicable."""
         result = super().session_info()
         store = Store()
-        self.env["res.users"]._init_store_data(store)
+        ResUsers = self.env["res.users"]
+        if cids := request.httprequest.cookies.get("cids", False):
+            allowed_company_ids = []
+            for company_id in [int(cid) for cid in cids.split("-")]:
+                if company_id in self.env.user.company_ids.ids:
+                    allowed_company_ids.append(company_id)
```

Comment:
We prefer to write list comprehensions for such cases
```py
allowed_company_ids = [
    company_id
    for company_id in map(int, cids.split('-'))
    if company_id in self.env.user.company_ids.ids
]
```
It is faster than a traditional for-loop (which are very slow in python) and also expected by python developers.

It's to late to change it, I'm just spreading the word of good practices for a future time.